c7n/actions.py

```diff
@@ -283,15 +304,28 @@ def get_groups(self, resources, metadata_key=None):
             elif remove_target_group_ids == 'all':
                 remove_groups = rgroups
             elif isinstance(remove_target_group_ids, list):
-                remove_groups = remove_target_group_ids
+                remove_groups = [r for r in remove_target_group_ids if r.startswith('sg-')]
```

Comment:
this method has hit a sane size limit, the group acquisition from the resource could move to a subclass responsibility for delegated resource type handling.